src/Q1.c

```diff
@@ -61,20 +64,34 @@ void * thread_func(void *arg){
     SetBit(places, place);
     pthread_mutex_unlock(&mut); 
 
-    // always some place available
-    printRegister(elapsedTime(), threadi, getpid(), pthread_self(), dur, place, ENTER);
+    // sending message with place to private fifo
+    char sendMessage[BUFSIZE];
+
+    if (elapsedTime() + dur <= nsecs) {
```

Comment:
Este if verifica se ao fim do tempo de uso a casa de banho ainda está aberta, caso esteja é que deixa entrar.
Pelo enunciado:
> 2LATE - servidor rejeita pedido por Quarto de Banho já ter encerrado

Ou seja, só manda o 2LATE se na altura do pedido a casa de banho estiver encerrada, não no final do uso
```suggestion
    if (elapsedTime() <= nsecs) {
```